internal/controller/controllers/agent_controller.go

```diff
@@ -134,6 +134,46 @@ func (r *AgentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	return result, nil
 }
 
+func (r *AgentReconciler) updateInstallerArgs(ctx context.Context, c *common.Cluster, host *common.Host, agent *aiv1beta1.Agent) error {
+
+	if agent.Spec.InstallerArgs == host.InstallerArgs {
+		r.Log.Debugf("Nothing to update, installer args were already set")
+		return nil
+	}
+
+	// InstallerArgs are saved in DB as string after unmarshalling of []string
+	// that operation removes all whitespaces between words
+	// in order to be able to validate that field didn't changed
+	// doing reverse operation
+	// If agent.Spec.InstallerArgs was not set but host.InstallerArgs was, we need to delete InstallerArgs
+	args := models.InstallerArgsParams{Args: []string{}}
+	if agent.Spec.InstallerArgs != "" {
+		err := json.Unmarshal([]byte(agent.Spec.InstallerArgs), &args.Args)
+		if err != nil {
+			msg := fmt.Sprintf("Fail to unmarshal installer args for host %s in cluster %s", agent.Name, c.Name)
+			r.Log.WithError(err).Errorf(msg)
+			return common.NewApiError(http.StatusBadRequest, errors.Wrapf(err, msg))
+		}
+	}
+
+	// as we marshalling same var or []string, there is no point to verify error on marshalling it
+	argsBytes, _ := json.Marshal(args.Args)
+	// we need to validate if the equal one more after marshalling
+	if string(argsBytes) == host.InstallerArgs {
```

Comment:
Consider moving this if statment to line 149 and avoid unnecessary operations (150-160 isn't relevant if nothing changed)
```suggestion
	if agent.Spec.InstallerArgs == host.InstallerArgs {
```

Reply:
i can't , if you will look more precisely and read description :( you will see that to make equal operation work i need to unmarshal and than marshal json script (as reverse operation for how we save it in DB) only in this case i can verify strings. Another option was just to unmarshal both strings but i prefer to do operation on the same object

Reply:
Got it,
Consider extracting this logic to a separate function `formatInstallerArgs` and call it from both places.